docs/content/docs/build/containers.md

```diff
@@ -143,14 +143,14 @@ container.on("disposed", () => {
 
 ### isDirty
 
-A container is considered **dirty** if it has local changes that have not yet been acknowledged by the service. Closing the container while `isDirty === true` will result in the loss of any operations that have not yet been acknowledged by the service. Closing the container while true will result in the loss of these local changes.
-
-You should always check the `isDirty` flag before closing the container or navigating away from the page.
+A container is considered **dirty** if it has local changes that have not yet been acknowledged by the service. You should always check the `isDirty` flag before closing the container or navigating away from the page. Closing the container while `isDirty === true` may result in the loss of operations that have not yet been acknowledged by the service.
 
 A container is considered dirty in the following cases:
 
-1. The container has local changes that have not yet been acknowledged by the service. These unacknowledged changes will be lost if the container is closed.
-1. There is no network connection while making changes to the container. These changes cannot be acknowledged by the service until the network connection is restored.
+1. The container has been created in the detached state, and either it has not been attached yet or it is in the process of being attached (container is in `attaching` state). Changes are not sent to the service until the container is *attached*. If container is closed prior to being attached, host may never know if the file was created or not.
```

Comment:
> Changes are not sent to the service until the container is *attached*

I'd remove this. I think people will not follow what you want to say here. If you really want to stress it, I'd say that "container may stay in dirty state when transitioning from attaching to attached state due to changes made in attaching state". But I do not know if we want to put all these complexities here.

Reply:
Yes, removed - that paragraph would make sense without that sentence as well. I like the idea of simplifying it.